src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -198,14 +198,16 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
                 LocalDataStoreInstance.LocalDataStore = keyStore;
             }
 
-            services = new DefaultServiceProvider();
+            grainFactory = new GrainFactory();
+            IDictionary<Type,object> singletons = new Dictionary<Type, object> { { typeof(IGrainFactory), grainFactory } };
+            services = new DefaultServiceProvider(singletons);
```

Comment:
@nehmebilal I think there's value in providing a way to not be forced to use a container if the developer doesn't want to. I'm still a little hesitant on the specifics, but having a very basic default provider is similar to what MVC, EF and some other libraries are doing. Nevertheless, we might be able to just reuse [DependencyInjection/ServiceProvider.cs](https://github.com/aspnet/DependencyInjection/blob/dev/src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs), by calling [serviceCollection.BuildServiceProvider()](https://github.com/aspnet/DependencyInjection/blob/15bda756314f70cce0f6025f6ef8defab3617cfa/src/Microsoft.Extensions.DependencyInjection/ServiceCollectionExtensions.cs#L10)